### PR TITLE
Break up list of commands

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun @ricmatsui 

The list of commands in the main readme is relatively long, unstructured, and hard to understand for beginners. This is an attempt at making it more readable by breaking it up along functional boundaries.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,15 +20,23 @@ and the [tutorial](documentation/tutorial.md) to get an idea for how it works.
...
-* [git extract](/documentation/commands/git-extract.md) - copy selected commits from the current branch into their own branch
-* [git hack](/documentation/commands/git-hack.md) - cut a new feature branch off the main branch
-* [git kill](/documentation/commands/git-kill.md) - remove an obsolete feature branch
-* [git prune-branches](/documentation/commands/git-prune-branches.md) - delete merged branches
-* [git pull-request](/documentation/commands/git-pull-request.md) - create a new pull request
-* [git repo](/documentation/commands/git-repo.md) - view the repository homepage
-* [git ship](/documentation/commands/git-ship.md) - deliver a completed feature branch
-* [git sync](/documentation/commands/git-sync.md) - update the current branch with all relevant changes
+
+__Workflow steps__
```
Comment (allewun) [inline README.md]: I think "steps" can be omitted

File: README.md [hunk 2]
```diff
@@ -20,15 +20,23 @@ and the [tutorial](documentation/tutorial.md) to get an idea for how it works.
...
+* [git hack](/documentation/commands/git-hack.md) - cuts a new up-to-date feature branch off the main branch
```
Comment (allewun) [inline README.md]: realphabetize the commands?

**Comment by charlierudolph** [comment]
I like a grouping such as

```
- Creating new feature branches
hack 
extract

- Staying in sync
sync
sync-fork

- Interacting with Bitbucket/Github
pull-request
repo

- Merging a feature branch
ship

- Cleaning up old feature branches
kill
prune-branches

- Help and Configuration
town
```

It results in more groups but I think breaks down the commands 

The workflow group shows the order in which you expect these to be used, but ordering the groups can also show that off.

File: README.md (hunk 2 above)
Comment (kevgo) [inline README.md]: I prefer having them in the order in which they are used here. That makes it more understandable imho.

File: README.md (hunk 1 above)
Comment (kevgo) [inline README.md]: Good idea. I renamed this to `Development Workflow`

**Comment by ricmatsui** [comment]
Looks good to me!

**Comment by kevgo** [comment]
Okay, I have created two branches that show both versions:
* KG version: https://github.com/Originate/git-town/tree/kg-documentation
* CR version: https://github.com/Originate/git-town/tree/kg-documentation-cr

I still like my version (slightly) better, mostly because it reads easier and feels easier to understand. "GT adds 2 new categories, each one with a handful of commands" feels less complicated than "GT adds 5 new categories of commands, each one with several commands". 

Thoughts?

**Comment by charlierudolph** [comment]
The only things I dislike about your version is that `git extract` doesn't fit well into the workflow and that we lose alpha order.

What about moving `git extract` to repository maintenance and alphabetizing the commands in repository maintenance?

That way the workflow steps are the typical steps you follow (and I definitely do), and repository maintenance are the less used steps which we alphabetize as you don't use them in any particular order.

I also think we should move the tutorial link right next to the workflow steps as thats what that document explains.

**Comment by kevgo** [comment]
I like that! I have pushed the changes. Where exactly do you want the tutorial link to go?

**Comment by kevgo** [comment]
Since we seem to be in agreement here, and this is just a documentation commit (where we don't need a super tight review process), I'm going to merge this in now. We can add the remaining suggestion (moving the tutorial link) in a separate PR. Thanks for the good feedback on this!

**Pull request closed without merging**